src/Evolution/Ringdown/MinimumAhCExcisionRadius.cpp

```diff
@@ -0,0 +1,362 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+#include "Evolution/Ringdown/MinimumAhCExcisionRadius.hpp"
+
+#include <array>
+#include <cstddef>
+#include <optional>
+#include <vector>
+
+#include "DataStructures/Matrix.hpp"
+#include "DataStructures/Tags/TempTensor.hpp"
+#include "DataStructures/Tensor/IndexType.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "Domain/BlockLogicalCoordinates.hpp"
+#include "Domain/CoordinateMaps/Distribution.hpp"
+#include "Domain/CoordsToDifferentFrame.hpp"
+#include "Domain/Creators/BinaryCompactObject.hpp"
+#include "Domain/Creators/Sphere.hpp"
+#include "Domain/Creators/TimeDependentOptions/BinaryCompactObject.hpp"
+#include "Domain/Creators/TimeDependentOptions/ExpansionMap.hpp"
+#include "Domain/Creators/TimeDependentOptions/RotationMap.hpp"
+#include "Domain/Creators/TimeDependentOptions/ShapeMap.hpp"
+#include "Domain/Creators/TimeDependentOptions/Sphere.hpp"
+#include "Domain/StrahlkorperTransformations.hpp"
+#include "IO/H5/Dat.hpp"
+#include "IO/H5/File.hpp"
+#include "IO/H5/VolumeData.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/IO/ReadSurfaceYlm.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/Strahlkorper.hpp"
+#include "NumericalAlgorithms/SphericalHarmonics/StrahlkorperFunctions.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/Serialization/Serialize.hpp"
+
+namespace evolution::Ringdown {
+double minimum_ahc_excision_radius(
+    const std::string& path_to_volume_data,
+    const std::string& volume_subfile_name,
+    const std::string& path_to_horizons_h5,
+    const std::string& surface_subfile_name,
+    const std::string& path_to_AhC_distorted_h5,
+    const std::vector<std::string>& AhC_distorted_subfile_names,
+    const size_t requested_number_of_times_from_end, double match_time,
+    double settling_timescale, double excision_A_radius,
+    double excision_B_radius, std::array<double, 3> excision_A_center,
+    std::array<double, 3> excision_B_center,
+    const std::optional<std::array<double, 3>>& exp_func_and_2_derivs,
+    const std::optional<std::array<double, 3>>&
+        exp_outer_bdry_func_and_2_derivs,
+    const std::optional<std::vector<std::array<double, 4>>>&
+        rot_func_and_2_derivs,
+    const std::optional<std::array<std::array<double, 3>, 3>>&
+        trans_func_and_2_derivs) {
+  // Read the AhC coefficients from the H5 file
+  const std::vector<ylm::Strahlkorper<Frame::Inertial>>& ahc_inertial =
+      ylm::read_surface_ylm<Frame::Inertial>(
+          path_to_horizons_h5, surface_subfile_name,
+          requested_number_of_times_from_end);
+
+  ylm::Strahlkorper<Frame::Inertial> ahc_inertial_at_match_time{};
+  std::vector<double> ahc_times{};
+  // Read the AhC times from the H5 file
```

Comment:
I suggest you move this block into the function `ylm::read_surface_ylm` because you usually need these times when reading strahlkorpers (e.g. in `strahlkorper_coefs_in_ringdown_distorted_frame`). For example, return a `std::vector<std::pair<double, ylm::Strahlkorper>>` from `ylm::read_surface_ylm`. (separate commit)